src/Agent.Plugins/PipelineCache/PipelineCacheServer.cs

```diff
@@ -156,12 +159,21 @@ public class PipelineCacheServer
             string targetDirectory,
             CancellationToken cancellationToken)
         {
+
             DownloadDedupManifestArtifactOptions options = DownloadDedupManifestArtifactOptions.CreateWithManifestId(
                 manifestId,
                 targetDirectory,
                 proxyUri: null,
                 minimatchPatterns: null);
-            return dedupManifestClient.DownloadAsync(options, cancellationToken);
+            using( Process tarring = new Process())
+            {
+                tarring.StartInfo.FileName = "tar";
+                tarring.StartInfo.Arguments = $"-xf - -C {targetDirectory}";
+                tarring.StartInfo.UseShellExecute = false;
+                tarring.StartInfo.RedirectStandardInput = true;
+                tarring.Start();
+                return dedupManifestClient.DownloadPipelineCacheAsync(options, cancellationToken, tarring.StandardInput.BaseStream);
```

Comment:
Download function should probably be async

Reply:
need to wait for tarring to complete

Reply:
need to close the base stream

Reply:
see https://github.com/johnterickson/PipelineCachingTar/blob/master/Program.cs

Reply:
probably need to redirect stdout/stderr to log errors